Makefile

```diff
@@ -564,15 +566,42 @@ endif
 	fi
 
 # Installation
+
+# $(call install_into,dir, a b, .c .d) copies {a,b}.{c,d} as dir/{a,b}.{c,d}
+# (if such files exists and after creating dir if needed).
+install_into = \
+  $(MKDIR) "$(1)" && \
+  for d in $(2); do \
+    for suf in $(3); do \
+      if test -f $$d$$suf; then $(INSTALL_DATA) $$d$$suf "$(1)"; fi; \
+    done; \
+  done \
```

Comment:
Shorter and faster:
```bash
install_into = \
  $(MKDIR) "$(1)" && \
  for suf in $(3); do \
    $(INSTALL_DATA) $(addsuffix $$suf,$(2)) "$(1)"; \
  done
```

Reply:
Personally I don't like `make` function much, I find they often make the code harder to read; but I don't have a very strong argument against using one here.

Reply:
Are you sure? When I try this, I get errors like `install: cannot stat 'parsing/parsetree.cmt': No such file or directory`.

Reply:
(this .cmt doesn't exist because there is no .ml, which is fine, we simply shouldn't fail trying to install it)

Reply:
Indeed, it is a bit more complicated than what I suggested.

Still I think it would be a good idea to avoid having one invocation per filename -- process creation is notoriously inefficient on some platforms (e.g. Cygwin).

Reply:
Ok. I couldn't find any filter-existing-files function, but test -f should not spawn, so we can probably filter file by file, and then do one big install call.

Reply:
After much struggle trying to understand how makefiles work, I have something that does one call to install per call to install_into.